docs/tutorial.md

```diff
@@ -3,217 +3,904 @@ id: tutorial
+      validationSchema={Yup.object({
+        firstName: Yup.string()
+          .min(15, 'Must be 15 characters or less')
+          .required('Required'),
+        lastName: Yup.string()
+          .min(20, 'Must be 20 characters or less')
+          .required('Required'),
+        email: Yup.string()
+          .email('Invalid email addresss`')
+          .required('Required'),
+      })}
+      onSubmit={(values, { setSubmitting }) => {
+        setTimeout(() => {
+          alert(JSON.stringify(values, null, 2));
+          setSubmitting(false);
+        }, 400);
+      }}
+    >
+      {formik => (
+        <form onSubmit={formik.handleSubmit}>
+          <label htmlFor="firstName">First Name</label>
+          <input id="firstName" {...getFieldProps('firstName')} />
+          {formik.touched.firstName && formik.errors.firstName ? (
+            <div>{formik.errors.firstName}</div>
+          ) : null}
+          <label htmlFor="lastName">Last Name</label>
+          <input id="lastName" {...getFieldProps('lastName')} />
+          {formik.touched.lastName && formik.errors.lastName ? (
+            <div>{formik.errors.lastName}</div>
+          ) : null}
+          <label htmlFor="email">Email Address</label>
+          <input id="email" {...getFieldProps('email')} />
+          {formik.touched.email && formik.errors.email ? (
+            <div>{formik.errors.email}</div>
+          ) : null}
+          <button type="submit">Submit</button>
+        </form>
+      )}
+    </Formik>
+  );
+};
+```
+
+As you can see above, we swapped out `useFormik()` hook and replaced it with the `<Formik>` component. The `<Formik>` accept a function as it's children (a.k.a. a render prop). It's argument is the _exact_ same object returned by `useFormik()` (in fact, `<Formik>` calls `useFormik()` internally!!). Thus, our form works the same as before, except now we can use new components to express ourselves in a more concise manner.
 
 ```jsx
-// EditUserDialog.js
 import React from 'react';
-import Dialog from 'MySuperDialog';
 import { Formik, Field, Form, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+
+const SignupForm = () => {
+  return (
+    <Formik
+      initialValues={{ firstName: '', lastName: '', email: '' }}
+      validationSchema={Yup.object({
+        firstName: Yup.string()
+          .min(15, 'Must be 15 characters or less')
+          .required('Required'),
+        lastName: Yup.string()
+          .min(20, 'Must be 20 characters or less')
+          .required('Required'),
+        email: Yup.string()
+          .email('Invalid email addresss`')
+          .required('Required'),
+      })}
+      onSubmit={(values, { setSubmitting }) => {
+        setTimeout(() => {
+          alert(JSON.stringify(values, null, 2));
+          setSubmitting(false);
+        }, 400);
+      }}
+    >
+      <Form>
+        <label htmlFor="firstName">First Name</label>
+        <Field id="firstName" type="text" />
```

Comment:
All of these `id`s should be `name`s, no?

Reply:
They should actually be both, because `label for={}` applies only to the id for accessibility purposes.